modules/core/core-git/src/main/java/org/eclipse/dirigible/core/git/GitConnector.java

```diff
@@ -90,6 +83,51 @@
 		this.git = new Git(repository);
 	}
 
+	@Override
+	public ProjectOriginUrls getOriginUrls() {
+		StoredConfig gitConfig = repository.getConfig();
+
+		String fetchUrl = gitConfig.getString("remote", "origin", "url");
+
+		String pushUrl;
+		pushUrl = gitConfig.getString("remote", "origin", "pushurl");
+		if (pushUrl == null) {
+			pushUrl = fetchUrl; // fallback to fetch url if no explicit pushurl is configured
+		}
+
+		return new ProjectOriginUrls(fetchUrl, pushUrl);
+	}
+
+	@Override
+	public String setFetchUrl(String fetchUrl) throws URISyntaxException, GitAPIException {
+		RemoteUrl remoteGit = RemoteUrl.getRemoteUrl(repository);
+		remoteGit.setUriType(RemoteSetUrlCommand.UriType.FETCH);
+		remoteGit.setRemoteName("origin");
+		URIish newUrl = new URIish(fetchUrl);
+		remoteGit.setRemoteUri(newUrl);
+		remoteGit.call();
+
+		StoredConfig gitConfig = repository.getConfig();
+		gitConfig.setString("remote", "origin", "url", fetchUrl);
+		String updatedURL = gitConfig.getString("remote", "origin", "url");
+		return updatedURL;
```

Comment:
Is this needed? Shouldn't the `setFetchUrl()` method be `void` instead of having return type `String`?

Reply:
Comment not addressed

Reply:
My intention was to have the result of set operation immediately available as the new value for the frontend, as well as making addition check that setting is really done. 
Anyway, I agree return value is not needed and will convert setter to void.